anchore_engine/db/entities/upgrade.py

```diff
@@ -1955,6 +1960,49 @@ def db_upgrade_013_014():
     upgrade_oauth_client_014()
 
 
+def remove_incorrect_github_vuln_data():
+    """
+    Github feed data was being processed under incorrect types 'github:os' and 'github:unknown'. This function removes
+    those vulnerability records synced by the policy engine because they are incorrect data
+    """
+    engine = anchore_engine.db.entities.common.get_engine()
+
+    namespace_names = "('github:os', 'github:unknown')"
+    to_delete = (
+        (ImagePackageVulnerability.__tablename__, "vulnerability_namespace_name"),
+        (VulnerableArtifact.__tablename__, "namespace_name"),
+        (FixedArtifact.__tablename__, "namespace_name"),
+        (Vulnerability.__tablename__, "namespace_name"),
+        (FeedGroupMetadata.__tablename__, "name"),
+    )
+
+    for table_name, col_name in to_delete:
+        log.err(
+            "Removing records from table {} where {} is {}".format(
+                table_name, col_name, namespace_names
+            ),
+        )
+
+        try:
+            engine.execute(
+                f"DELETE from {table_name} WHERE {col_name} IN {namespace_names}"
+            )
+        except Exception as e:
+            err_string = (
+                f"Failed to perform deletion on {table_name} - exception: {str(e)}"
+            )
+            log.err(err_string)
+            raise Exception(err_string)
```

Comment:
I would recommend using SQLAlchemy ORM to delete the records - refer to archive_data_upgrade_005_006 or other functions that use session object to perform db ops. Direct SQL execution should be sparingly used only when the functionality is not available via SQLAlchemy such as altering tables - adding, dropping, updating columns, creating indices etc. Using the ORM would also help with cleaning up foreign key relationships with cascading deletes (delete records from Vulnerability) and prevent typos in column names